packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -1122,6 +1106,20 @@ export function TransferPanel() {
         return TransferPanelMainErrorMessage.GAS_ESTIMATION_FAILURE
 
       case 'success': {
+        const requiredGasFees: number = (() => {
+          // For SC wallets, the relayer pays the gas fees
+          if (isSmartContractWallet) {
+            if (isDepositMode) {
+              // L2 fee is paid in callvalue and still needs to come from the SC wallet for retryable cost estimation to succeed
+              return gasSummary.estimatedL2GasFees
+            }
+
+            return 0
+          }
+
+          return gasSummary.estimatedL1GasFees + gasSummary.estimatedL2GasFees
+        })()
```

Comment:
now the same snippet is copied and pasted

can't we revert to the previous solution but without `gasSummary.estimatedTotalGasFees`?

unless it helps with future refactoring

Reply:
It helps with future refactoring that's why I started this PR